### PR TITLE
Add proxy-server current connections metric.

### PR DESCRIPTION
**Pull request by jkh52**

### CODE REVIEW

File: pkg/server/metrics/metrics.go [hunk 1]
```diff
@@ -54,16 +56,41 @@ func newServerMetrics() *ServerMetrics {
 		},
 		[]string{},
 	)
+	connections := prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "grpc_connections",
+			Help:      "Number of current grpc connections, partitioned by type.",
```
Comment (caesarxuchao) [inline pkg/server/metrics/metrics.go:64]: s/by type/by service method

File: pkg/server/metrics/metrics.go [hunk 2]
```diff
@@ -22,11 +22,12 @@ import (
...
 )
 
-type Direction string
-
 const (
 	namespace = "konnectivity_network_proxy"
 	subsystem = "server"
+
+	Frontend = "frontend"
+	Backend = "backend"
```
Comment (caesarxuchao) [inline pkg/server/metrics/metrics.go:30]: How about using the actual method names (connect/proxy)?

Frontend/Backend can be confusing.

**Review by caesarxuchao — COMMENTED** [review]
Nits one names. Otherwise lgtm.

File: pkg/server/metrics/metrics.go (hunk 1 above)
Comment (jkh52) [reply pkg/server/metrics/metrics.go:64]: Done.

File: pkg/server/metrics/metrics.go (hunk 2 above)
Comment (jkh52) [reply pkg/server/metrics/metrics.go:30]: Good idea, done.

Do you think it's worth adding service name as well, for completeness? (We only have one method per service though.)

Comment (caesarxuchao) [reply pkg/server/metrics/metrics.go]: I think the chance of us adding new methods of the same name is low, so "connect"/"proxy" are probably good enough.

**Comment by caesarxuchao** [comment]
/lgtm

Omitting the "approval" to give @Jefftree or @cheftako a chance to take a look.

File: pkg/server/metrics/metrics.go [hunk 3]
```diff
@@ -54,16 +59,40 @@ func newServerMetrics() *ServerMetrics {
...
 	prometheus.MustRegister(latencies)
-	return &ServerMetrics{latencies: latencies}
+	prometheus.MustRegister(connections)
+	return &ServerMetrics{latencies: latencies, connections: connections}
 }
 
 // Reset resets the metrics.
 func (a *ServerMetrics) Reset() {
 	a.latencies.Reset()
+	a.connections.Reset()
```
Comment (cheftako) [inline pkg/server/metrics/metrics.go:82]: Might be missing something but I didn't think one usually reset a gauge.

Comment (jkh52) [reply pkg/server/metrics/metrics.go]: Thanks, fixed.

I don't see any calls to metrics.Metrics.Reset(), is it called implicitly somewhere that I don't realize?

Comment (cheftako) [reply pkg/server/metrics/metrics.go]: Great question. I'm not a prometheus expert but if this is wired in properly I would think prometheus would call it (directly or indirectly) when it grabbed the metrics. Reset is used to stop double counting on counters as they are cumulative results.

**Pull request merged**